plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/RunAppEngineShortcutTest.java

```diff
@@ -85,10 +81,8 @@ public void testRunAppEngine_hiddenEvenIfAppEngineProjectFileIsOpen() throws Cor
   private static boolean appEngineMenuExists(IProject project) {
     SWTBotTreeItem selected = SwtBotProjectActions.selectProject(
         new SWTWorkbenchBot(), project.getName());
-    Menu runAsMenu = selected.contextMenu("Run As").widget.getMenu();
-    Matcher<MenuItem> matcher = WidgetMatcherFactory.withRegex("App Engine");
-
-    List<MenuItem> menuItems = new MenuFinder().findMenus(runAsMenu, matcher, false);
-    return !menuItems.isEmpty();
+    List<String> menuItems = selected.contextMenu("Run As").menuItems();
+    Predicate<String> isAppEngineMenu = name -> name.contains("App Engine");
+    return menuItems.stream().anyMatch(isAppEngineMenu);
```

Comment:
The SWTBot solution can be optimized too.  Might want to contrast the streams to:
```
try {
   selected.contextMenu("Run As").menu(WidgetMatcherFactory.withRegex("App Engine"), false /* recursive */, 0 /* index */);
   return true;
} catch (WidgetNotFoundException ex) {
   return false;
}
```
Which has a slight advantage in that it will retry if the menu generation is expensive?

Reply:
The problem with `.menu(...)` is that it will wait for a long time until it throws `WidgetNotFoundException`:

```java
	public SWTBotMenu menu(Matcher<MenuItem> matcher, final boolean recursive, final int index) throws WidgetNotFoundException {
		WaitForObjectCondition<MenuItem> waitForMenuItem = Conditions.waitForMenuItem(this, matcher, recursive, index);
		new SWTBot().waitUntilWidgetAppears(waitForMenuItem);
		return new SWTBotMenu(waitForMenuItem.get(0), matcher);
	}
```

So it works for finding an existing menu, but it takes too long to verify an item doesn't exist.

Reply:
I remember that is why I went for `selected.contextMenu("Run As").widget.getMenu()` previously.